src/kokkos_ext/ArborX_DetailsKokkosExtClassLambda.hpp

```diff
@@ -0,0 +1,25 @@
+/****************************************************************************
+ * Copyright (c) 2017-2023 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_DETAILS_KOKKOS_EXT_CLASS_LAMBDA
+#define ARBORX_DETAILS_KOKKOS_EXT_CLASS_LAMBDA
+
+#include <Kokkos_Macros.hpp>
+
+#if KOKKOS_VERSION >= 40000
```

Comment:
```C++
#ifdef KOKKOS_CLASS_LAMBDA
#define ARBORX_CLASS_LAMBDA KOKKOS_CLASS_LAMBDA
#else
...

Reply:
This may be ok as long as we have a comment to remove it with 4.0.0.

Reply:
I thought about doing that but just as Andrey said I find it easier for us to spot it when we require Kokkos 4.0.0.

Reply:
I am not blocking but I don't think being able to grep for `KOKKOS_VERSION` is a good argument.  You can have a comment `// remove when KOKKOS_VERSION >= 40000 is alway true`

Reply:
You seem to feel stronger about it than me so I'll just change.